src/core/model_accessing_API.jl

```diff
@@ -27,6 +31,14 @@ function add_agent_to_model!(agent::AbstractAgent, model::VecABM)
     return
 end
 
+function add_agent_to_model!(agent::AbstractAgent, model::StructABM)
+    agent.id != nagents(model) + 1 && error(lazy"Cannot add agent of ID $(agent.id) in a vector ABM of $(nagents(model)) agents. Expected ID == $(nagents(model)+1).")
+    tuple_agent = NamedTuple(x => getfield(agent, x) for x in fieldnames(typeof(agent)))
+    push!(agent_container(model), tuple_agent)
+    extra_actions_after_add!(tuple_agent, model)
```

Comment:
```suggestion
    extra_actions_after_add!(model[agent.id], model)
```

this should be like this (at least is already a bit better) because you pass the wrapper this way